test/note_display.test.js

```diff
@@ -0,0 +1,530 @@
+    const pendingPin = await this.createNote();
+    await this.itemManager.changeItem(
+      pendingPin.uuid,
+      (mutator) => {
+        mutator.pinned = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includePinned = false;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(1);
+  });
+
+  it('includePinned on', async function () {
+    await this.createNote();
+    const pendingPin = await this.createNote();
+    await this.itemManager.changeItem(
+      pendingPin.uuid,
+      (mutator) => {
+        mutator.pinned = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includePinned = true;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(2);
+  });
+
+  it('includeTrashed off', async function () {
+    await this.createNote();
+    const pendingTrash = await this.createNote();
+    await this.itemManager.changeItem(
+      pendingTrash.uuid,
+      (mutator) => {
+        mutator.trashed = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includeTrashed = false;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(1);
+  });
+
+  it('includeTrashed on', async function () {
+    await this.createNote();
+    const pendingTrash = await this.createNote();
+    await this.itemManager.changeItem(
+      pendingTrash.uuid,
+      (mutator) => {
+        mutator.trashed = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includeTrashed = true;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(2);
+  });
+
+  it('includeArchived off', async function () {
+    await this.createNote();
+    const pendingArchive = await this.createNote();
+    await this.itemManager.changeItem(
+      pendingArchive.uuid,
+      (mutator) => {
+        mutator.archived = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includeArchived = false;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(1);
+  });
+
+  it('includeArchived on', async function () {
+    await this.createNote();
+    const pendingArchive = await this.createNote();
+    await this.itemManager.changeItem(
+      pendingArchive.uuid,
+      (mutator) => {
+        mutator.archived = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includeArchived = true;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(2);
+  });
+
+  it('protectedSearchEnabled false', async function () {
+    const normal = await this.createNote();
```

Comment:
Having `'world'` visible somewhere in the creation steps would make the test easier to read in isolation, what do you think?